### PR TITLE
Add data support for next open hours

### PR DESCRIPTION
**Pull request by Sti3bas**
This PR fixes `Call to a member function toDateTime() on array` error when trying to call `nextOpen` method with:
```php
'monday' => [
    [
        'hours' => '09:00-11:00',
        'data' => ['foobar'],
    ],
    '13:00-19:00',
],
```

### CODE REVIEW

File: src/OpeningHoursForDay.php [hunk 1]
```diff
@@ -93,7 +93,11 @@ public function nextClose(Time $time)
 
     protected function findNextOpenInWorkingHours(Time $time, TimeRange $timeRange)
     {
-        if ($timeRange->containsTime($time) && next($timeRange) !== $timeRange) {
+        if ($timeRange->containsTime($time) && $next = next($timeRange) !== $timeRange) {
```
Comment (kylekatarnls) [inline src/OpeningHoursForDay.php]: Due to operators precedence, `$next` will always be a boolean (in this case, `false`), so `! $next instanceof Time` will always be `true`. We are just always skipping a value here as far as I understand. I will try to create a unit test to confirm it.

**Comment by kylekatarnls** [comment]
Thanks for your proposal. We will check it and get back in touch soon.

File: src/OpeningHoursForDay.php (hunk 1 above)
Comment (Sti3bas) [reply src/OpeningHoursForDay.php]: Yes, you are right.. the strangest thing is that if you just return false from this method all tests will still pass.

```php
protected function findNextOpenInWorkingHours(Time $time, TimeRange $timeRange)
{
    return false;
}
```

**Comment by kylekatarnls** [comment]
Thanks, with new changes, no this pass without error. Please try to update to `dev-master` to see if it solves your problem.

**Comment by Sti3bas** [comment]
Yes, `dev-master` solves the problem. Thank you!

**Comment by kylekatarnls** [comment]
Released in https://github.com/spatie/opening-hours/releases/tag/2.1.0-beta.1

**Pull request merged**